addons/crm/models/crm_lead.py

```diff
@@ -292,7 +292,8 @@ def _compute_company_id(self):
             # propose a new company based on responsible, limited by team
             if not proposal:
                 if lead.user_id:
-                    proposal = lead.team_id.company_id or lead.user_id.company_id
+                    user_company = lead.user_id.company_id & self.env.companies
+                    proposal = lead.team_id.company_id or user_company
```

Comment:
A bit strange to compute user_company first, then do an or with another value ... Can't we make it explicit like
```
if lead.user_id and lead.team_id.company_id:
    proposal = lead.team_id.company_id
elif lead.user_id and lead.user_id.company_id < self.env.companies:
    proposal = lead.user_id.company_id
elif lead:team_id:
    [BLAH]
```

Reply:
Comment: not sure my proposal is right actually, seems not the same use case. But you got the idea, I don't think we sholud have "and" and "or" within a condition, it makes the understanding of flow difficult to follow.